openghg/retrieve/_search.py

```diff
@@ -605,11 +605,29 @@ def _base_search(**kwargs: Any) -> SearchResults:
             logger.exception(msg=error_msg)
             raise ValueError(error_msg)
 
-        # Here we create a dictionary of the metadata keyed by the Datasource UUID
-        # we'll create a pandas DataFrame out of this in the SearchResult object
-        # for better printing / searching within a notebook
+        # Due to versions of OpenGHG < 0.6.0 not storing the object store
+        # in the metadata we need to pass that information in here for backwards
+        # compatibility.
         metadata = {r["uuid"]: r for r in metastore_records}
 
+        updated_metadata = {}
+        store_data = {"object_store": bucket}
+
+        for uid, _metadata in metadata.items():
+            if "object_store" in _metadata:
+                ds_bucket = _metadata["object_store"]
+                if ds_bucket != bucket:
+                    raise ObjectStoreError(
+                        f"Mismatch between Datasource object_store: {ds_bucket} and "
```

Comment:
I'm a little cautious about adding in an error here that users won't be able to do anything about - any ideas why this error might come in and what advice we could give to fix this?